src/components/policy/policy_regular/src/policy_helper.cc

```diff
@@ -366,28 +398,32 @@ bool CheckAppPolicy::operator()(const AppPoliciesValueType& app_policy) {
       "Permissions for application:" << app_id << " have been changed.");
 
   if (IsPredefinedApp(app_policy)) {
-    for (const policy_table::ApplicationPolicies::value_type& app :
-         snapshot_->policy_table.app_policies_section.apps) {
-      if (app_policy.first == app.second.get_string()) {
-        if (RESULT_CONSENT_NOT_REQIURED != result) {
-          SetPendingPermissions(app, result);
-          NotifySystem(app);
-        }
-        SendPermissionsToApp(app.first, app_policy.second.groups);
-      }
+    const auto& snapshot_app_policy_begin =
+        snapshot_->policy_table.app_policies_section.apps.begin();
+    const auto& snapshot_app_policy_end =
+        snapshot_->policy_table.app_policies_section.apps.end();
+
+    auto find_app = [&app_id](AppPoliciesValueType app) {
+      return app_id == app.second.get_string();
+    };
+
+    const auto& app = std::find_if(
+        snapshot_app_policy_begin, snapshot_app_policy_end, find_app);
+
+    if ((snapshot_app_policy_end != app) &&
+        (RESULT_CONSENT_NOT_REQIURED != result)) {
```

Comment:
`RESULT_CONSENT_NOT_REQIURED` -> `RESULT_CONSENT_NOT_REQUIRED`

Reply:
@jacobkeeler fixed in 76277de2068688672613bec5ac92a1beae5574dd